core/src/main/scala/kafka/coordinator/GroupMetadata.scala

```diff
@@ -145,6 +146,7 @@ private[coordinator] class GroupMetadata(val groupId: String, initialState: Grou
   private val members = new mutable.HashMap[String, MemberMetadata]
   private val offsets = new mutable.HashMap[TopicPartition, OffsetAndMetadata]
   private val pendingOffsetCommits = new mutable.HashMap[TopicPartition, OffsetAndMetadata]
+  private val deletedPartitions = new mutable.HashSet[TopicPartition]
```

Comment:
Trying to think of a better name for this. Haha, this is verbose, but accurate: `deletedPartitionsWithPendingOffsetCommits`. An alternative for the sake of discussion is to try to reuse the `pendingOffsetCommits` collection to handle partition deletion. For example, maybe we make the logic something like the following:

1. On deletion, remove the topic partition from `pendingOffsetCommits` blindly.
2. On offset commit completion, only insert the offset commit into `offsets` if there is _some_ entry in `pendingOffsetCommits`, and only delete from `pendingOffsetCommits` if the entry matches.

What do you think?